stacks-common/src/types/chainstate.rs

```diff
@@ -247,6 +247,13 @@ impl_array_hexstring_fmt!(StacksBlockId);
 impl_byte_array_newtype!(StacksBlockId, u8, 32);
 impl_byte_array_serde!(StacksBlockId);
 
+pub struct TenureBlockId(pub StacksBlockId);
```

Comment:
Can you remove this, please?  There's nothing about this type that requires it to identify a tenure-start block.  It's just a type alias (with extra steps) for `StacksBlockId`.

Reply:
I added this to get some type-safety, to ensure that a tenure-change block is passed to several of the methods, or get a compile time error.

Reply:
But there's no guarantee that `TenureBlockId` is a tenure-change block.  You can cast _any_ `StacksBlockId` to a `TenureBlockId`, and the code will compile.

Reply:
Right, but you have to actually think about it to do that cast.

Reply:
Whereas without this new type, you could just pass any block and not realize that it should've been a tenure-change block. I'll add some documentation on this type (that should've been there).

Reply:
Docstring added in eb9b3a182.